avm/res/resources/deployment-script/main.bicep

```
@@ -0,0 +1,274 @@
+metadata name = 'Deployment Scripts'
+metadata description = 'This module deploys Deployment Scripts.'
+metadata owner = 'Azure/module-maintainers'
+
+// ================ //
+// Parameters       //
+// ================ //
+@description('Required. Name of the Deployment Script.')
+@maxLength(24)
+param name string
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Required. Specifies the Kind of the Deployment Script.')
+@allowed([
+  'AzureCLI'
+  'AzurePowerShell'
+])
+param kind string
+
+@metadata({
+  example: '''
+  {
+    '/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.ManagedIdentity/userAssignedIdentities/myID': {}
+    '/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.ManagedIdentity/userAssignedIdentities/myID2': {}
+  }
+  '''
+})
+@description('Optional. The managed identity definition for this resource.')
+param managedIdentities managedIdentitiesType
+
+@description('Optional. Resource tags.')
+param tags object?
+
+@description('Optional. Azure PowerShell module version to be used. See a list of supported Azure PowerShell versions: https://mcr.microsoft.com/v2/azuredeploymentscripts-powershell/tags/list.')
+param azPowerShellVersion string?
+
+@description('Optional. Azure CLI module version to be used. See a list of supported Azure CLI versions: https://mcr.microsoft.com/v2/azure-cli/tags/list.')
+param azCliVersion string?
+
+@description('Optional. Script body. Max length: 32000 characters. To run an external script, use primaryScriptURI instead.')
+param scriptContent string?
+
+@description('Optional. Uri for the external script. This is the entry point for the external script. To run an internal script, use the scriptContent instead.')
+param primaryScriptUri string?
+
+@metadata({
+  example: '''
+secureList: [
+  {
+    name: 'string'
+    secureValue: 'string'
+    value: 'string'
+  }
+]
+'''
+})
+@description('Optional. The environment variables to pass over to the script. The list is passed as an object with a key name "secureList" and the value is the list of environment variables (array). The list must have a \'name\' and a \'value\' or a \'secretValue\' property for each object.')
+@secure()
+param environmentVariables object = {}
+
+@description('Optional. List of supporting files for the external script (defined in primaryScriptUri). Does not work with internal scripts (code defined in scriptContent).')
+param supportingScriptUris array?
+
+@metadata({
+  example: '''
+[
+  {
+    id: '/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.Network/virtualNetworks/myVnet/subnets/mySubnet'
+  }
+]
+'''
+})
+@description('Optional. List of subnet IDs to use for the container group. This is required if you want to run the deployment script in a private network.')
+param subnetIds array?
```

```suggestion
param subnetResourceIds array?
```
We have written somewhere in the specs that we should always try to be explicit when mentioning `Id`, even if it diverts from the API. So in this case we should use `resourceId` explicitely. For those it's of course relativ straight forward, for a e.g. a `identityId` parameter it is a lot more imporant to tell if its the principal or client or resource id.